rpcserver.go

```diff
@@ -1530,6 +1530,20 @@ func (r *rpcServer) SendPayment(paymentStream lnrpc.Lightning_SendPaymentServer)
 					htlcSema <- struct{}{}
 				}()
 
+				// Calculate the user-specified fee cut-off
+				// using the FeeCutOff oneof field.
+				var maxFee *btcutil.Amount = nil
+				switch nextPayment.GetFeeCutOff().(type) {
+				case *lnrpc.SendRequest_LimitSatoshis:
+					*maxFee = btcutil.Amount(nextPayment.GetLimitSatoshis())
+				case *lnrpc.SendRequest_LimitRatio:
+					*maxFee = btcutil.Amount(nextPayment.GetLimitRatio()) * amt
```

Comment:
Should I do any more checking here on the limit ratio (e.g. check if it is negative)? As it stands now I think it will simply fail to find a route.